### DISCUSSION TITLE
chamando funções dentro de uma requisição - duvidas

### DISCUSSION THREAD

**Discussion by AlexhaifaGarcia** [HashLoad/horse#390] [General]
Boa Tarde amigos, 

Tenho uma dúvida a respeito de requisições e chamadas de funções.
Sendo cada requisição uma thread, tenho a seguinte dúvida.

Minha requisição de consulta faz a chamada de uma função onde nesta função é criado um TSQLconnection e um TSQLQuery, eu executo e depois destruo os componentes. 

Eu gostaria de saber se vou ter algum tipo de problema com requisições simultaneas que chamam função ou funções ( caso venho a precisar mais de uma ) de dentro da minha requisição. 

Muito obrigado pelo apoio.

**Comment by viniciussanchez** [comment]
Boa tarde... não terá problemas se cada uma delas criar e destruir suas próprias instâncias. O que não pode é usar recursos compartilhados. Por exemplo, várias funções usando a mesma instância de um TFDQuery. Assim em quanto uma estaria montando um SQL, a outra estaria dando um clear por exemplo.
